php-checks/src/main/java/org/sonar/php/checks/IgnoredReturnValueCheck.java

```diff
@@ -0,0 +1,294 @@
+    "array_unique",
+    "array_values",
+    "array",
+    "compact",
+    "count",
+    "current",
+    "in_array",
+    "key_exists",
+    "key",
+    "pos",
+    "range",
+    "sizeof",
+
+    // Variable handling Functions
+    "boolval",
+    "debug_zval_dump",
+    "doubleval",
+    "empty",
+    "floatval",
+    "get_defined_vars",
+    "get_resource_type",
+    "gettype",
+    "intval",
+    "is_array",
+    "is_bool",
+    "is_double",
+    "is_float",
+    "is_int",
+    "is_integer",
+    "is_iterable",
+    "is_long",
+    "is_null",
+    "is_numeric",
+    "is_object",
+    "is_real",
+    "is_resource",
+    "is_scalar",
+    "is_string",
+    "isset",
+    "serialize",
+    "strval",
+    "unserialize",
+
+    // Classes/Object Functions
+    "get_called_class",
+    "get_class_methods",
+    "get_class_vars",
+    "get_class",
+    "get_declared_classes",
+    "get_declared_interfaces",
+    "get_declared_traits",
+    "get_object_vars",
+    "get_parent_class",
+    "interface_exists",
+    "is_a",
+    "is_subclass_of",
+    "method_exists",
+    "property_exists",
+    "trait_exists",
+
+    // Ctype Functions
+    "ctype_alnum",
+    "ctype_alpha",
+    "ctype_cntrl",
+    "ctype_digit",
+    "ctype_graph",
+    "ctype_lower",
+    "ctype_print",
+    "ctype_punct",
+    "ctype_space",
+    "ctype_upper",
+    "ctype_xdigit",
+
+    // Filter Functions
+    "filter_has_var",
+    "filter_id",
+    "filter_input_array",
+    "filter_input",
+    "filter_list",
+    "filter_var_array",
+    "filter_var",
+
+    // Function handling Functions
+    "func_get_arg",
+    "func_get_args",
+    "func_num_args",
+    "function_exists"
+  ));
+
+  @Override
+  public void visitExpressionStatement(ExpressionStatementTree tree) {
+    checkExpression(tree.expression());
+    super.visitExpressionStatement(tree);
+  }
+
+  @Override
+  public void visitForStatement(ForStatementTree tree) {
+    tree.init().forEach(this::checkExpression);
+    tree.update().forEach(this::checkExpression);
+    super.visitForStatement(tree);
+  }
+
+  private void checkExpression(ExpressionTree expression) {
+    while (expression instanceof ParenthesisedExpressionTree) {
+      expression = ((ParenthesisedExpressionTree) expression).expression();
+    }
+    if (expression.is(Tree.Kind.FUNCTION_CALL)) {
+      ExpressionTree callee = ((FunctionCallTree) expression).callee();
+      String name = nameOf(callee);
+      checkName(callee, name);
+    } else if (expression.is(Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+      ArrayInitializerFunctionTree initializer = (ArrayInitializerFunctionTree) expression;
+      checkName(initializer, "array");
+    }
+  }
+
+  private void checkName(Tree issueLocation, @Nullable String name) {
+    if (name != null && PURE_FUNCTIONS.contains(name)) {
+      context().newIssue(this, issueLocation, "The return value of \"" + name + "\" must be used.");
+    }
+  }
+
+  @Nullable
+  private static String nameOf(Tree tree) {
```

Comment:
Probably worth to have it in `CheckUtils`. I have seen ugly parts in other checks where `toString()` is used. It would also allow proper testing.